### PR TITLE
Improve installation instructions in the README

### PR DESCRIPTION
**Pull request by kouroshD**
@GiulioRomualdi  @traversaro  
Can you please check the updates for the dependencies installations, and if it is fine merge it? thanks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,10 +21,128 @@ It implements the following architecture:
...
+              
+             
+      - Configure, generate the cmake (cmake gui --> robotology yarp) and build the yarp using vs15.
+
+
+6. **Cyberith SDK:**   To allow the virtualizer module to capture the operator data, the virtualizer needs two modules to download: the CybSDK_app (used for the calibration!) and the CybSDK:
+
+   1. `CybSDK_app`: you can download the application from [this link](https://developer.cyberith.com/downloads). Go to 'Tools' tab, download the 'Virtualizer Control Panel'. Copy the download application to the workspace. Everytime start using the virtualizer, please calibrate it using this application.
+     
+   2. `CybSDK`: you can download the application from [this link](https://developer.cyberith.com/downloads). Go to 'SDK' tab, download the 'C++ SDK for Windows (Developed for: Visual Studio 2015 Community)'. Copy the download application to the workspace. in theory, we should build it in our Windows machine, but time being it is already installed and is working, so you do not need to install it.
```
Comment (traversaro) [inline README.md]: I am not getting the last sentence to be honest. It is already installed were? Consider that this document is a generic installation documentation, so it is not safe to refer to the state of a specific setup.

File: README.md [hunk 2]
```diff
@@ -21,10 +21,128 @@ It implements the following architecture:
...
+   
+      - Add the following varibales to the Rapid Evironment Edittor:
```
[the remark this answers is not in the document]
Comment (traversaro) [reply README.md]: I would rephrase: 
~~~
Append the following diretory to the `Path` User environmental variable, for example using the [Rapid Environment Editor](https://www.rapidee.com): 
~~~
(but probably it make sense to just mention the Rapid Enviroment Editor once in all the README

File: README.md [hunk 3]
```diff
@@ -21,10 +21,128 @@ It implements the following architecture:
 
 
 # :page_facing_up: Dependencies
-* [YARP](http://www.yarp.it/): to handle the comunication with the robot with both ovrheadset and SDLjoypad drivers;
-* [Walking-controllers](https://github.com/robotology/walking-controllers) to handl the hand cartesian task and walking commands
-* [Oculus SDK](https://developer.oculus.com/downloads/package/oculus-sdk-for-windows/): The native Windows SDK.
-* [Cyberith SDK](https://www.cyberith.com/research-development/): To allow the virtualizer module to capture the operator data.
+1. **robotology super-build:** Install the robotology super-build on a Windows machine as described here [Windows installation notes](https://github.com/robotology/robotology-superbuild#windows):
```
[the remark this answers is not in the document]
Comment (kouroshD) [reply README.md]: I have updated it, mentioning superbuild as optional point, and one can install YARP, as mentioned on its official website.

File: README.md (hunk 1 above)
Comment (kouroshD) [reply README.md]: I have updated the sentence, I was trying to say there is already the '.dll' binary, so they do not need to build it again.

File: README.md [hunk 4]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+
+              `LibOVR_LibOVR_Extras_INCLUDE_DIR`: '<path to codes workspace>/OculusSDK/LibOVR/Include/Extras'
+
+              `LibOVR_LibOVR_INCLUDE_DIR:PATH`: '<path to codes workspace>/OculusSDK/LibOVR/Include'
+
+              `LibOVR_LibOVR_LIBRARY_DEBUG`: '<path to codes workspace>/OculusSDK/LibOVR/Lib/Windows/x64/Debug/VS2015/LibOVR.lib'
+
+              `LibOVR_LibOVR_LIBRARY_RELEASE`: '<path to codes workspace>/OculusSDK/LibOVR/Lib/Windows/x64/Release/VS2015/LibOVR.lib'
+
+              `YARP_USE_LOBOVR`: check the box
```
Comment (traversaro) [inline README.md]:
```suggestion
               `YARP_USE_LIBOVR`: check the box
```

**Comment by traversaro** [comment]
Ok for me, probably is better to wait for @GiulioRomualdi as the current mantainer for merging the PR.

File: README.md [hunk 5]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
 
 
 # :page_facing_up: Dependencies
-* [YARP](http://www.yarp.it/): to handle the comunication with the robot with both ovrheadset and SDLjoypad drivers;
-* [Walking-controllers](https://github.com/robotology/walking-controllers) to handl the hand cartesian task and walking commands
-* [Oculus SDK](https://developer.oculus.com/downloads/package/oculus-sdk-for-windows/): The native Windows SDK.
-* [Cyberith SDK](https://www.cyberith.com/research-development/): To allow the virtualizer module to capture the operator data.
+ 1. **robotology superbuild [optional]:** Install the robotology superbuild on a Windows machine as described here [Windows installation notes](https://github.com/robotology/robotology-superbuild#windows):
+
+    1. opencv: Do not use the opencv package as mentioned there (Dec 2018), instead install it from the [opencv website] (https://opencv.org/releases.html), version 3.4.4;
```
Comment (GiulioRomualdi) [inline README.md]: Can you explain better ```Do not use the opencv package as mentioned there (Dec 2018)```, You can just add the link at the following issue: https://github.com/robotology/robotology-superbuild/issues/145  
Furthermore there is  there is a space between **]** and **(** ```[opencv website] (https://opencv.org/releases.html)```

File: README.md [hunk 6]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+
+   + if you don't want to use superbuild for installation, skip it and go directly to second dependency (YARP)
+2. **YARP:** Using superbuild install the [YARP](http://www.yarp.it/): to handle the comunication with the robot with both ovrheadset and SDLjoypad drivers;
+
+   + if you are not using superbuild, look at [this link](http://www.yarp.it/) for installing YARP.
+
+3.  **Walking-controllers:** In order to teleoperate the robot, this package should be installed in another workstation;
```
Comment (GiulioRomualdi) [inline README.md]: Actually, you do not need to install the walking controller inside the windows machine.

File: README.md [hunk 7]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+
+4. In the Super-build, enable the QT5 and Eigen3 to install (follow the instructions there).
```
Comment (GiulioRomualdi) [inline README.md]: Which instructions?

File: README.md [hunk 8]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+
+5. **Oculus SDK:** In order to install the oculus SDK, yolu need to install first the Oculus app on windows (oculus application), install the depenecies(GLEW, GLFW3), and finally install the sdk (LibOVR):
```
Comment (GiulioRomualdi) [inline README.md]: Typo `yolu` :smile:

File: README.md [hunk 9]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+
+   1. `oculus application`: Download and install the oculus [setup application](https://www.oculus.com/setup/). It is used for both the Oculus Virtual Reality headset (ovrheadset) and joypads (touch controller).
+          
+   2. `GLEW`: Dowload the version 2.1.0 of the [glew libarary](http://glew.sourceforge.net/index.html) from this[link](https://sourceforge.net/projects/glew/files/glew/2.1.0/) (if you want to dowload from the [glew libarary](http://glew.sourceforge.net/index.html) select the source file)
+     
+      - extract and copy the glew library to your robot/code workspace (in our case in same path of robotology-superbuild)
+      
+      - the library has the makfiles, so go to the 'glew\glew-2.1.0\glew-2.1.0\build\vc12' and using the VS15 (VS 2017) build the library (release and debug!)
```
Comment (GiulioRomualdi) [inline README.md]: Typo `makfiles`

File: README.md [hunk 10]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+              `GLEW_INCLUDE_DIR`: '<path to codes workspace>/glew/glew-2.1.0/glew-2.1.0/include
+               (inside this folder you should find the GL folder and inside that the header files)
+          
+              `GLEW_LIBRARY_DEBUG`: '<path to codes workspace>/glew/glew-2.1.0/glew-2.1.0/lib/Debug/x64/glew32d.lib'
+          
+              `GLEW_LIBRARY_RELEASE`: '<path to codes workspace>/glew/glew-2.1.0/glew-2.1.0/lib/Release/x64/glew32.lib'
+              
+              `YARP_USE_GLEW`: check the box
+         
+      - Add the following values to the given variables of the Rapid Environment Editor:
```
Comment (GiulioRomualdi) [inline README.md]: I think you can avoid referring to _Rapid Environment Editor_.

File: README.md [hunk 11]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+      - Add the following varibales to the Rapid Evironment Edittor:
+             
+              `GLFW3_DIR`= '<path to codes workspace>\glfw\glfw-3.2.1\glfw-3.2.1\build\install' (variable type is String)
+             
+      - Don't forget to configure, generate the cmake (cmake gui --> robotology yarp) and build release mode the yarp using vs15.  
+             
+      - if you have problems with compiling GLFW, [this link](https://www.glfw.org/docs/latest/compile_guide.html#compile_generate) or [here](https://github.com/nigels-com/glew) may help.
+          
+          
+    4. `LibOVR`: to dowload the SDK, you can go to [this website](), choose "Native Windows", then "Core Package: OCULUS SDK for windows". select the version 1.16.0 to download (you can install the latest one as well 1.32 (the last release) but there are some breaking changes in the APIs in version 1.17 and you will probably need to patch the driver software). To download you can go easily to [link](https://developer.oculus.com/downloads/package/oculus-sdk-for-windows/1.16.0/) to download the sdk version 1.16.0.
```
Comment (GiulioRomualdi) [inline README.md]: `LibOVR: to dowload the SDK, you can go to [this website]()`. Link missing. + Typo (dowload).

Since the version 1.32 is not compatible with YARP you can directly write: 

4. `LibOVR`: to download the SDK, you can go to [this website](), choose "Native Windows", then "Core Package: OCULUS SDK for windows" and select the version 1.16.0. Or easily follow this [link](https://developer.oculus.com/downloads/package/oculus-sdk-for-windows/1.16.0/).

File: README.md [hunk 12]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+   
+      - Extract and place the package in your workspace (in our case in same path of robotology-superbuild), 1inside that there are two libraries which we need: LibOVR, and LibOVRKernel.
```
Comment (GiulioRomualdi) [inline README.md]: typo `1inside`

File: README.md [hunk 13]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+      
+      - Build (both release and debug) the projects inside the sdk using VS15: 
+        
+        '\LibOVR\Projects\Windows\VS2015\LibOVR.vcxproj'
+        
+        '\LibOVRKernel\Projects\Windows\VS2015\LibOVRKernel.vcxproj '
```
Comment (GiulioRomualdi) [inline README.md]: Please use \` \` environment when you specify a path (e.g.  '\LibOVR\Projects\Windows\VS2015\LibOVR.vcxproj'
 becomes `\LibOVR\Projects\Windows\VS2015\LibOVR.vcxproj`)

File: README.md [hunk 14]
```diff
@@ -21,10 +21,131 @@ It implements the following architecture:
...
+      - Configure, generate the cmake (cmake gui --> robotology yarp) and build the yarp using vs15.
+
+
+6. **Cyberith SDK:**   To allow the virtualizer module to capture the operator data, the virtualizer needs two modules to download: the CybSDK_app (used for the calibration!) and the CybSDK:
+
+   1. `CybSDK_app`: you can download the application from [this link](https://developer.cyberith.com/downloads). Go to 'Tools' tab, download the 'Virtualizer Control Panel'. Copy the download application to the workspace. Everytime start using the virtualizer, please calibrate it using this application.
+     
+   2. `CybSDK`: you can download the application from [this link](https://developer.cyberith.com/downloads). Go to 'SDK' tab, download the 'C++ SDK for Windows (Developed for: Visual Studio 2015 Community)'. Copy the download application to the workspace. in theory, we should build it in our Windows machine, but time being it is already built (there exists the .dll file) and is working, so you do not need to build it.
+   
+      - Add the following variables to the Rapid Evironment Editor:
```
Comment (GiulioRomualdi) [inline README.md]: Like before you can avoid mentioning `Rapid Evironment Editor`

File: README.md (hunk 4 above)
Comment (GiulioRomualdi) [inline README.md]: you can remove white lines between   `LibOVR_LibOVR_INCLUDE_DIR:PATH: '<path to codes workspace>/OculusSDK/LibOVR/Include'` and   `LibOVR_LibOVR_LIBRARY_DEBUG: '<path to codes ...` (etc)

File: README.md (hunk 14 above)
Comment (GiulioRomualdi) [reply README.md]: Or if you want follow @traversaro suggestion https://github.com/robotology/walking-teleoperation/pull/9#discussion_r243104349

File: README.md (hunk 5 above)
Comment (kouroshD) [reply README.md]: done.

File: README.md (hunk 6 above)
Comment (kouroshD) [reply README.md]: yes, you are right, I have updated it.

File: README.md (hunk 7 above)
Comment (kouroshD) [reply README.md]: added!

File: README.md (hunk 10 above)
Comment (kouroshD) [reply README.md]: I have updated it, but I kept the software we have used as an option!

File: README.md (hunk 12 above)
Comment (kouroshD) [reply README.md]: so many typos!

File: README.md (hunk 4 above)
Comment (kouroshD) [reply README.md]: I think its readability is higher with a line space in between!

**Comment by kouroshD** [comment]
Thank you @GiulioRomualdi for the careful comments. I have updated it. Please have a look again. 
Also, I have added a gitignore for temporary files!

**Comment by GiulioRomualdi** [comment]
Well done @kouroshD! We can merge the PR

**Comment by GiulioRomualdi** [comment]
rebase and merging
